gateway/src/main/java/org/georchestra/gateway/autoconfigure/app/LanguageSupportWebFilter.java

```diff
@@ -0,0 +1,43 @@
+package org.georchestra.gateway.autoconfigure.app;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.event.EventListener;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.web.server.ServerWebExchange;
+import org.springframework.web.server.WebFilter;
+import org.springframework.web.server.WebFilterChain;
+import org.springframework.web.server.adapter.DefaultServerWebExchange;
+import org.springframework.web.server.adapter.HttpWebHandlerAdapter;
+import reactor.core.publisher.Mono;
+
+public class LanguageSupportWebFilter implements WebFilter {
+
+    @Value("${language}")
+    private String language;
+    private final ApplicationContext applicationContext;
+    private HttpWebHandlerAdapter httpWebHandlerAdapter;
+
+    public LanguageSupportWebFilter(final ApplicationContext applicationContext) {
+        this.applicationContext = applicationContext;
+    }
+
+    @EventListener(ApplicationReadyEvent.class)
+    public void loadHttpHandler() {
+        this.httpWebHandlerAdapter = applicationContext.getBean(HttpWebHandlerAdapter.class);
+    }
+
+    @Override
+    public Mono<Void> filter(final ServerWebExchange exchange, final WebFilterChain chain) {
+        if (language == null || language.equals("")) {
+            language = "en";
+        }
+        final ServerHttpRequest httpRequest = exchange.getRequest().mutate()
+                .headers(httpHeaders -> httpHeaders.set("Accept-Language", language)).build();
```

Comment:
I'm wondering if the value like the default one ("en") is legit. Checking in my web developer console, my browser is sending the following header:

```
accept-language: en-US,en;q=0.9,fr;q=0.8,de;q=0.7
```

MDN also states that the syntax is more complex: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language

Reply:
Other thought: isn't this header sent by the browser already ?

Reply:
Yes, but language parameter should be set in default.properties file, but user can force it using lang parameter on the request, if language is not set on default.prop or force by user, then we well read web navigator configuration (sent always for every request).
The question here is why I'm using header, this is because spring webflux  using header based configuration for local, so I have intercepted request and for header set to make it works.

Reply:
Regarding  header format 'accept-language: en-US,en;q=0.9,fr;q=0.8,de;q=0.7'
Both works, if you set 'en' or 'fr' it works , because new Locale('en') and new Locale('en-US','en') both works.
https://docs.oracle.com/javase/tutorial/i18n/locale/create.html